app/src/main/java/com/duckduckgo/app/httpsupgrade/api/HttpsUpgradeListDownloader.kt

```diff
@@ -38,21 +51,47 @@ class HttpsUpgradeListDownloader @Inject constructor(
             val call = service.https()
             val response = call.execute()
 
-            if (response.isCached && httpsUpgradeDao.count() != 0) {
+            if (response.isCached && dbWriteStatusStore.hasWriteCompleted() && httpsUpgradeDao.count() != 0) {
```

Comment:
Couple of points:

* Probably don't need the count check?

* Also, might be more elegant to store the etag once all the records are inserted and compare the etag in the response vs the one stored?